

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>References &mdash; MDAnalysis User Guide  documentation</title>
  

  
  
    <link rel="shortcut icon" href="_static/mdanalysis-logo.ico"/>
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
        <script type="text/javascript" src="https://unpkg.com/@jupyter-widgets/html-manager@^0.18.0/dist/embed-amd.js"></script>
        <script type="text/javascript" src="_static/js/versions.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="search" type="application/opensearchdescription+xml"
          title="Search within MDAnalysis User Guide  documentation"
          href="_static/opensearch.xml"/>

  
  <link rel="stylesheet" href="_static/css/msmb.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Tests in MDAnalysis" href="testing.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >
          

          
            <a href="index.html">
          

          
            
            <img src="_static/user_guide.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                2.0.0-dev0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Getting started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/quickstart.html">Quick start guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="faq.html">Frequently asked questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/README.html">Examples</a></li>
</ul>
<p class="caption"><span class="caption-text">Data structures</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="universe.html">Universe</a></li>
<li class="toctree-l1"><a class="reference internal" href="atomgroup.html">AtomGroup</a></li>
<li class="toctree-l1"><a class="reference internal" href="groups_of_atoms.html">Groups of atoms</a></li>
<li class="toctree-l1"><a class="reference internal" href="selections.html">Atom selection language</a></li>
<li class="toctree-l1"><a class="reference internal" href="topology_system.html">The topology system</a></li>
</ul>
<p class="caption"><span class="caption-text">Trajectories</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="trajectories/trajectories.html">Trajectories</a></li>
<li class="toctree-l1"><a class="reference internal" href="trajectories/slicing_trajectories.html">Slicing trajectories</a></li>
<li class="toctree-l1"><a class="reference internal" href="trajectories/transformations.html">On-the-fly transformations</a></li>
<li class="toctree-l1"><a class="reference internal" href="units.html">Units and constants</a></li>
</ul>
<p class="caption"><span class="caption-text">Input/output</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="reading_and_writing.html">Reading and writing files</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats/index.html">Format overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats/guessing.html">Guessing</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats/auxiliary.html">Auxiliary files</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats/selection_exporters.html">Selection exporters</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats/format_reference.html">Format reference</a></li>
</ul>
<p class="caption"><span class="caption-text">Analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html">Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#alignments-and-rms-fitting">Alignments and RMS fitting</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#distances-and-contacts">Distances and contacts</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#trajectory-similarity">Trajectory similarity</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#structure">Structure</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#volumetric-analyses">Volumetric analyses</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#dimension-reduction">Dimension reduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/README.html#polymers-and-membranes">Polymers and membranes</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/custom_trajectory_analysis.html">Writing your own trajectory analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="examples/analysis/custom_parallel_analysis.html">How to run your analysis parallelly</a></li>
</ul>
<p class="caption"><span class="caption-text">Advanced</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="standard_selections.html">Standard residues in MDAnalysis selections</a></li>
<li class="toctree-l1"><a class="reference internal" href="advanced_topology.html">Advanced topology concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="datasets.html">Example data</a></li>
</ul>
<p class="caption"><span class="caption-text">Contributing</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="contributing.html">Contributing to MDAnalysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="contributing_code.html">Contributing to the main codebase</a></li>
<li class="toctree-l1"><a class="reference internal" href="contributing_docs.html">Contributing to the user guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="preparing_releases_and_hotfixes.html">Preparing a release</a></li>
<li class="toctree-l1"><a class="reference internal" href="module_imports.html">Module imports in MDAnalysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="testing.html">Tests in MDAnalysis</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">References</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#citations-using-duecredit">Citations using Duecredit</a></li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">MDAnalysis User Guide</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>References</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/references.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="references">
<span id="id1"></span><h1>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h1>
<p>MDAnalysis and the included algorithms are scientific software that
are described in academic publications. <strong>Please cite these papers when you use
MDAnalysis in published work.</strong></p>
<p>It is possible to <a class="reference external" href="https://www.mdanalysis.org/docs/documentation_pages/references.html#citations-using-duecredit" title="(in MDAnalysis v1.0)"><span class="xref std std-ref">automatically generate a list of references</span></a> for any program that uses
MDAnalysis. This list (in common reference manager formats) contains
the citations associated with the specific algorithms and libraries
that were used in the program.</p>
<div class="section" id="citations-using-duecredit">
<h2>Citations using Duecredit<a class="headerlink" href="#citations-using-duecredit" title="Permalink to this headline">¶</a></h2>
<p>Citations can be automatically generated using <a class="reference external" href="https://github.com/duecredit/duecredit">duecredit</a>, depending on the
packages used. Duecredit is easy to install via <code class="docutils literal notranslate"><span class="pre">pip</span></code>. Simply type:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>pip install duecredit
</pre></div>
</div>
<p><a class="reference external" href="https://github.com/duecredit/duecredit">duecredit</a> will remain an optional dependency, i.e. any code using
MDAnalysis will work correctly even without duecredit installed.</p>
<p>A list of citations for <code class="docutils literal notranslate"><span class="pre">yourscript.py</span></code> can be obtained using simple
commands.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span> /path/to/yourmodule
python -m duecredit yourscript.py
</pre></div>
</div>
<p>or set the environment variable <span class="target" id="index-0"></span><code class="xref std std-envvar docutils literal notranslate"><span class="pre">DUECREDIT_ENABLE</span></code></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>DUECREDIT-ENABLE<span class="o">=</span>yes python yourscript.py
</pre></div>
</div>
<p>Once the citations have been extracted (to a hidden file in the
current directory), you can use the <strong class="program">duecredit</strong> program to
export them to different formats. For example, one can display them in
BibTeX format, using:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>duecredit summary --format<span class="o">=</span>bibtex
</pre></div>
</div>
<p><strong>Please cite your use of MDAnalysis and the packages and algorithms
that it uses. Thanks!</strong></p>
<p id="bibtex-bibliography-references-0"><dl class="citation">
<dt class="bibtex label" id="beckstein-zipping-2009"><span class="brackets">BDPW09</span></dt>
<dd><p>Oliver Beckstein, Elizabeth J. Denning, Juan R. Perilla, and Thomas B. Woolf. Zipping and Unzipping of Adenylate Kinase: Atomistic Insights into the Ensemble of Open↔Closed Transitions. <em>Journal of Molecular Biology</em>, 394(1):160–176, November 2009. 00107. URL: <a class="reference external" href="https://linkinghub.elsevier.com/retrieve/pii/S0022283609011164">https://linkinghub.elsevier.com/retrieve/pii/S0022283609011164</a>, <a class="reference external" href="https://doi.org/10.1016/j.jmb.2009.09.009">doi:10.1016/j.jmb.2009.09.009</a>.</p>
</dd>
<dt class="bibtex label" id="best-native-2013"><span class="brackets">BHE13</span></dt>
<dd><p>R. B. Best, G. Hummer, and W. A. Eaton. Native contacts determine protein folding mechanisms in atomistic simulations. <em>Proceedings of the National Academy of Sciences</em>, 110(44):17874–17879, October 2013. 00259. URL: <a class="reference external" href="http://www.pnas.org/cgi/doi/10.1073/pnas.1311599110">http://www.pnas.org/cgi/doi/10.1073/pnas.1311599110</a>, <a class="reference external" href="https://doi.org/10.1073/pnas.1311599110">doi:10.1073/pnas.1311599110</a>.</p>
</dd>
<dt class="bibtex label" id="franklin-minactionpath-2007"><span class="brackets">FKDD07</span></dt>
<dd><p>Joel Franklin, Patrice Koehl, Sebastian Doniach, and Marc Delarue. MinActionPath: maximum likelihood trajectory for large-scale structural transitions in a coarse-grained locally harmonic energy landscape. <em>Nucleic Acids Research</em>, 35(suppl_2):W477–W482, July 2007. 00083. URL: <a class="reference external" href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkm342">https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkm342</a>, <a class="reference external" href="https://doi.org/10.1093/nar/gkm342">doi:10.1093/nar/gkm342</a>.</p>
</dd>
<dt class="bibtex label" id="gowers-mdanalysis-2016"><span class="brackets">GLB+16</span></dt>
<dd><p>Richard J. Gowers, Max Linke, Jonathan Barnoud, Tyler J. E. Reddy, Manuel N. Melo, Sean L. Seyler, Jan Domański, David L. Dotson, Sébastien Buchoux, Ian M. Kenney, and Oliver Beckstein. MDAnalysis: A Python Package for the Rapid Analysis of Molecular Dynamics Simulations. <em>Proceedings of the 15th Python in Science Conference</em>, pages 98–105, 2016. 00152. URL: <a class="reference external" href="https://conference.scipy.org/proceedings/scipy2016/oliver_beckstein.html">https://conference.scipy.org/proceedings/scipy2016/oliver_beckstein.html</a>, <a class="reference external" href="https://doi.org/10.25080/Majora-629e541a-00e">doi:10.25080/Majora-629e541a-00e</a>.</p>
</dd>
<dt class="bibtex label" id="hall-characterization-2007"><span class="brackets">HKP+07</span></dt>
<dd><p>Benjamin A. Hall, Samantha L. Kaye, Andy Pang, Rafael Perera, and Philip C. Biggin. Characterization of Protein Conformational States by Normal-Mode Frequencies. <em>Journal of the American Chemical Society</em>, 129(37):11394–11401, September 2007. 00020. URL: <a class="reference external" href="https://doi.org/10.1021/ja071797y">https://doi.org/10.1021/ja071797y</a>, <a class="reference external" href="https://doi.org/10.1021/ja071797y">doi:10.1021/ja071797y</a>.</p>
</dd>
<dt class="bibtex label" id="janin-conformation-1978"><span class="brackets">JWLM78</span></dt>
<dd><p>Joël Janin, Shoshanna Wodak, Michael Levitt, and Bernard Maigret. Conformation of amino acid side-chains in proteins. <em>Journal of Molecular Biology</em>, 125(3):357 – 386, 1978. 00874. URL: <a class="reference external" href="http://www.sciencedirect.com/science/article/pii/0022283678904084">http://www.sciencedirect.com/science/article/pii/0022283678904084</a>, <a class="reference external" href="https://doi.org/10.1016/0022-2836(78)90408-4">doi:10.1016/0022-2836(78)90408-4</a>.</p>
</dd>
<dt class="bibtex label" id="liu-fast-2009"><span class="brackets">LAT09</span></dt>
<dd><p>Pu Liu, Dimitris K. Agrafiotis, and Douglas L. Theobald. Fast determination of the optimal rotational matrix for macromolecular superpositions. <em>Journal of Computational Chemistry</em>, pages n/a–n/a, 2009. URL: <a class="reference external" href="http://doi.wiley.com/10.1002/jcc.21439">http://doi.wiley.com/10.1002/jcc.21439</a>, <a class="reference external" href="https://doi.org/10.1002/jcc.21439">doi:10.1002/jcc.21439</a>.</p>
</dd>
<dt class="bibtex label" id="lovell-structure-2003"><span class="brackets">LDA+03</span></dt>
<dd><p>Simon C. Lovell, Ian W. Davis, W. Bryan Arendall, Paul I. W. de Bakker, J. Michael Word, Michael G. Prisant, Jane S. Richardson, and David C. Richardson. Structure validation by Cα geometry: ϕ,ψ and Cβ deviation. <em>Proteins: Structure, Function, and Bioinformatics</em>, 50(3):437–450, January 2003. 03997. URL: <a class="reference external" href="http://doi.wiley.com/10.1002/prot.10286">http://doi.wiley.com/10.1002/prot.10286</a>, <a class="reference external" href="https://doi.org/10.1002/prot.10286">doi:10.1002/prot.10286</a>.</p>
</dd>
<dt class="bibtex label" id="michaud-agrawal-mdanalysis-2011"><span class="brackets">MADWB11</span></dt>
<dd><p>Naveen Michaud-Agrawal, Elizabeth J. Denning, Thomas B. Woolf, and Oliver Beckstein. MDAnalysis: A toolkit for the analysis of molecular dynamics simulations. <em>Journal of Computational Chemistry</em>, 32(10):2319–2327, July 2011. 00778. URL: <a class="reference external" href="http://doi.wiley.com/10.1002/jcc.21787">http://doi.wiley.com/10.1002/jcc.21787</a>, <a class="reference external" href="https://doi.org/10.1002/jcc.21787">doi:10.1002/jcc.21787</a>.</p>
</dd>
<dt class="bibtex label" id="nguyen-nglviewinteractive-2018"><span class="brackets">NCR18</span></dt>
<dd><p>Hai Nguyen, David A Case, and Alexander S Rose. NGLview–interactive molecular graphics for Jupyter notebooks. <em>Bioinformatics</em>, 34(7):1241–1242, April 2018. 00024. URL: <a class="reference external" href="https://academic.oup.com/bioinformatics/article/34/7/1241/4721781">https://academic.oup.com/bioinformatics/article/34/7/1241/4721781</a>, <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btx789">doi:10.1093/bioinformatics/btx789</a>.</p>
</dd>
<dt class="bibtex label" id="seyler-path-2015"><span class="brackets">SKTB15</span></dt>
<dd><p>Sean L. Seyler, Avishek Kumar, M. F. Thorpe, and Oliver Beckstein. Path Similarity Analysis: A Method for Quantifying Macromolecular Pathways. <em>PLOS Computational Biology</em>, 11(10):e1004568, October 2015. URL: <a class="reference external" href="https://dx.plos.org/10.1371/journal.pcbi.1004568">https://dx.plos.org/10.1371/journal.pcbi.1004568</a>, <a class="reference external" href="https://doi.org/10.1371/journal.pcbi.1004568">doi:10.1371/journal.pcbi.1004568</a>.</p>
</dd>
<dt class="bibtex label" id="smart-pore-1993"><span class="brackets">SGW93</span></dt>
<dd><p>O S Smart, J M Goodfellow, and B A Wallace. The pore dimensions of gramicidin A. <em>Biophysical Journal</em>, 65(6):2455–2460, December 1993. 00522. URL: <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1225986/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1225986/</a>, <a class="reference external" href="https://doi.org/10.1016/S0006-3495(93)81293-1">doi:10.1016/S0006-3495(93)81293-1</a>.</p>
</dd>
<dt class="bibtex label" id="smart-hole-1996"><span class="brackets">SNW+96</span></dt>
<dd><p>O. S. Smart, J. G. Neduvelil, X. Wang, B. A. Wallace, and M. S. Sansom. HOLE: a program for the analysis of the pore dimensions of ion channel structural models. <em>Journal of Molecular Graphics</em>, 14(6):354–360, 376, December 1996. 00935. <a class="reference external" href="https://doi.org/10.1016/s0263-7855(97)00009-x">doi:10.1016/s0263-7855(97)00009-x</a>.</p>
</dd>
<dt class="bibtex label" id="stelzl-flexible-2014"><span class="brackets">SFSB14</span></dt>
<dd><p>Lukas S. Stelzl, Philip W. Fowler, Mark S. P. Sansom, and Oliver Beckstein. Flexible gates generate occluded intermediates in the transport cycle of LacY. <em>Journal of Molecular Biology</em>, 426(3):735–751, February 2014. 00000. URL: <a class="reference external" href="https://asu.pure.elsevier.com/en/publications/flexible-gates-generate-occluded-intermediates-in-the-transport-c">https://asu.pure.elsevier.com/en/publications/flexible-gates-generate-occluded-intermediates-in-the-transport-c</a>, <a class="reference external" href="https://doi.org/10.1016/j.jmb.2013.10.024">doi:10.1016/j.jmb.2013.10.024</a>.</p>
</dd>
<dt class="bibtex label" id="theobald-rapid-2005"><span class="brackets">The05</span></dt>
<dd><p>Douglas L. Theobald. Rapid calculation of RMSDs using a quaternion-based characteristic polynomial. <em>Acta Crystallographica Section A Foundations of Crystallography</em>, 61(4):478–480, July 2005. 00127. URL: <a class="reference external" href="http://scripts.iucr.org/cgi-bin/paper?S0108767305015266">http://scripts.iucr.org/cgi-bin/paper?S0108767305015266</a>, <a class="reference external" href="https://doi.org/10.1107/S0108767305015266">doi:10.1107/S0108767305015266</a>.</p>
</dd>
<dt class="bibtex label" id="tiberti-encore-2015"><span class="brackets">TPB+15</span></dt>
<dd><p>Matteo Tiberti, Elena Papaleo, Tone Bengtsen, Wouter Boomsma, and Kresten Lindorff-Larsen. ENCORE: Software for Quantitative Ensemble Comparison. <em>PLOS Computational Biology</em>, 11(10):e1004415, October 2015. 00031. URL: <a class="reference external" href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004415">https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004415</a>, <a class="reference external" href="https://doi.org/10.1371/journal.pcbi.1004415">doi:10.1371/journal.pcbi.1004415</a>.</p>
</dd>
</dl>
</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="testing.html" class="btn btn-neutral float-left" title="Tests in MDAnalysis" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019-2020, Lily Wang, Irfan Alibay, Rocco Meli, Richard J. Gowers, and Oliver Beckstein.

    </p>
  </div> 

</footer>

        </div>
      </div>

    </section>

  </div>
  <script>
    var versions_json_url = 'https://userguide.mdanalysis.org/versions.json'
</script>

<div class="rst-versions" data-toggle="rst-versions" role="note"
     aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      <span class="fa fa-book"></span>
        
      <span class="fa fa-caret-down"></span>
    </span>

    <div class="rst-other-versions">
        <dl id="versionselector">
            <dt>Other Versions</dt>
        </dl>

    </div>
</div>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>